### PR TITLE
Add file Extension to mediaType mapping in MI Registry

### PR DESCRIPTION
**Pull request by arunans23**
Resolves https://github.com/wso2/micro-integrator/issues/175

Since micro-integrator registry is different from WSO2 registry, it does not have all the metadata related to the registry files. In Script mediator, it requires a [DataHandler](https://github.com/wso2/wso2-synapse/blob/master/modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java#L494) which in turn requires [media type](https://github.com/wso2/carbon-mediation/blob/master/components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/WSO2Registry.java#L704) of the uploaded registry file. As a temporary fix, we need to map all the file extensions with Content-Type so that we can retrieve the media-type in runtime by looking at the file extension.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java [hunk 1]
```diff
@@ -947,4 +959,48 @@ public Properties getResourceProperties(String entryKey) {
...
     }
+
+    /**
+     * Populate file extension to media type mapping
+     *
+     * @return Map which contains file extension to media type mapping
+     */
+    private Map<String, String> createFileExtensionsMap() {
+
+        Map<String, String> extensionContentTypeMap = new HashMap();
```
Comment (a5anka) [inline components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java]:
```suggestion
        Map<String, String> extensionContentTypeMap = new HashMap<>();
```

File: components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java [hunk 2]
```diff
@@ -815,14 +822,19 @@ private OMNode readNonXML (URL url) throws IOException {
                 return null;
             }
 
-            StringBuilder strBuilder = new StringBuilder();
-            try (BufferedReader bReader = new BufferedReader(new InputStreamReader(inputStream))) {
-                String line;
-                while ((line = bReader.readLine()) != null) {
-                    strBuilder.append(line);
+            if (mediaType == "" || mediaType == "text/plain") {
```
[the remark this answers is not in the document]
Comment (arunans23) [reply components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java]: Fixed in https://github.com/wso2/carbon-mediation/pull/1251

File: components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java [hunk 3]
```diff
@@ -947,4 +959,48 @@ public Properties getResourceProperties(String entryKey) {
...
+     */
+    private String lookUpFileMediaType(String fileUrl) {
+        String extension = "";
+        int i = fileUrl.lastIndexOf('.');
+        if (i > 0) {
+            extension = fileUrl.substring(i);
+        }
+
+        String mediaType = "";
+        if (fileExtensionMediaTypeMap != null && fileExtensionMediaTypeMap.containsKey(extension)) {
```
Comment (sdkottegoda) [inline components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java]: Can the map be null at this point? It's initialized in the init method

Comment (arunans23) [reply components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java]: A redundant check. This code is changed in the latest branch.

**Pull request merged**